### PR TITLE
Updating to arti 1.0.1

### PR DESCRIPTION
**Pull request by ineiti**
Updating to the latest version of the arti code.

### CODE REVIEW

File: src/ffi/ios.rs [hunk 1]
```diff
@@ -20,6 +20,7 @@ pub unsafe extern "C" fn logger_init() {
         .with_ansi(false)
         .finish();
     tracing::subscriber::set_global_default(subscriber).expect("to be the only logger");
+    tracing::info!("Version 0.4.6");
```
Comment (tharvik) [inline src/ffi/ios.rs:23]: debug?

File: src/flatfiledirmgr.rs [hunk 2]
```diff
@@ -260,6 +261,10 @@ impl<R: Runtime> NetDirProvider for FlatFileDirMgr<R> {
     fn events(&self) -> BoxStream<'static, DirEvent> {
         Box::pin(self.tx_events.subscribe())
     }
+
+    fn params(&self) -> Arc<(dyn AsRef<NetParameters> + 'static)> {
```
Comment (tharvik) [inline src/flatfiledirmgr.rs:265]:
```suggestion
    fn params(&self) -> Arc<dyn AsRef<NetParameters>> {
```

File: Cargo.toml [hunk 3]
```diff
@@ -25,25 +25,25 @@ categories = [
 crate-type = ["cdylib", "lib", "staticlib"]
 
 [dependencies]
-arti-client = { version = "0.5", default-features = false, features = [
+arti-client = { version = "0.7", default-features = false, features = [
```
Comment (tharvik) [inline Cargo.toml:28]: can you also bump the other deps while at it?

Comment (ineiti) [reply Cargo.toml:28]: They are bumped - `arti` is quite strange when it comes to versioning. If you look at

https://gitlab.torproject.org/tpo/core/arti/-/blob/main/crates/arti-client/Cargo.toml

you can see that all other dependencies from arti, `tor-*`, are all the same version as what I have here. I made an issue some time ago to complain that some patch-versions broke the dependencies. They said it shouldn't, but then did it again...

So we gotta live with it.

File: src/ffi/ios.rs (hunk 1 above)
Comment (ineiti) [reply src/ffi/ios.rs]: yes, sorry. But I _would_ like to have something in there that references which version it's using. Would be really helpful for debugging...

File: src/flatfiledirmgr.rs (hunk 2 above)
Comment (ineiti) [reply src/flatfiledirmgr.rs]: Auto-complete by IntelliJ... Same thing happens in Visual Studio Code. Must be some cargo-tool.

File: Cargo.toml (hunk 3 above)
Comment (tharvik) [reply Cargo.toml:28]: hum, too bad, surely it'll be more stable as the first stable CLI has been released; ho well..

I meant the other deps, such as `postage` and `jni`, you can use [`cargo outdated -d1`](https://github.com/kbknapp/cargo-outdated) to know about availbe updates

**Review by tharvik — DISMISSED** [review]

File: Cargo.toml (hunk 3 above)
Comment (ineiti) [reply Cargo.toml:28]: One more cargo tool :)

Unrelated question: why do the cargo-tools always propose to install with `--locked`?

**Pull request merged**